Add C3 language

michabay05 opened the pull request:
<!--- Briefly describe your changes in the field above. -->

## Description
<!--- If necessary, go into depth of what this pull request is doing. -->
Fix #6072 

## Checklist:
- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      <!-- Replace FOOBAR with the new extension, and KEYWORDS with keywords unique to the language. Repeat for each extension added. -->
      -  https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.c3+fn
      - https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.c3i+fn
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - `argsparse.c3`: [Source](https://github.com/joshring/argparse/blob/main/src/argsparse.c3)
    - Sample license(s):
      - `argsparse.c3`: MIT License
  - [x] I have included a syntax highlighting grammar: https://github.com/c3lang/vscode-c3/tree/main
      <!-- Setting a color is strongly recommended, but optional: `#cccccc` is used by default -->
  - [x] I have added a color
    - Hex value: `#2563eb`
    - Rationale: This color is one of the two primary colors from the C3 logo.
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

JoelKipper commented:
LGTM!

lildude commented on lib/linguist/languages.yml:
```diff
@@ -860,6 +860,16 @@ C2hs Haskell:
+  color: "#969696"
+  extensions:
+  - ".c3"
+  - ".c3i"
+  - ".c3l"
```
All extensions being added need to meet our usage requirements and must have samples. If any one extension has lower user than others, it'll hold back the entire PR until usage is sufficient.

lildude commented on lib/linguist/languages.yml:
```diff
@@ -860,6 +860,16 @@ C2hs Haskell:
+  - ".c3"
+  - ".c3i"
+  - ".c3l"
+  tm_scope: source.c
```
This should be the scope of the grammar you're adding, else why add a grammar?

lildude commented on lib/linguist/languages.yml:
```diff
@@ -860,6 +860,16 @@ C2hs Haskell:
+  - ".c3i"
+  - ".c3l"
+  tm_scope: source.c
+  ace_mode: c_cpp
```
If this language is an evolution of C, wouldn't it make more sense to use C's ace mode, assuming C3 isn't supported?

lildude reviewed (changes requested):
See inline comments. Additionally, 

- you've added two unnecesary submodules; remove them
- your samples don't look like real world illustrations; replace them with real use examples and update the PR templatre
- you've not updated the search query in the PR template; please do so. This is your chance to prove your language meets our usage requirements.

tareifz commented:
>       * [x]  I have added a color
>         
>         * Hex value: `#969696`
>         * Rationale: Since this language is intended to be an evolution of the C programming language, the chosen color is close to the color used for `C` `#555555`.


why the colors are not the same as the language logo?

#2563eb
#7c3aed

michabay05 commented:
> > ```
> >   * [x]  I have added a color
> >     
> >     * Hex value: `#969696`
> >     * Rationale: Since this language is intended to be an evolution of the C programming language, the chosen color is close to the color used for `C` `#555555`.
> > ```
> 
> I suggest using one of the colors of the language logo?
> 
> `#2563eb` `#7c3aed`

Updated color (I don't have a strong position any color can be chosen)

rudy-in commented:
check out my PR
if its any useful you can close yours :D

rudy-in commented:
i did it out of annoyance
C3 is very popular but not on github i was mad

rudy-in commented:
@michabay05 by any chance are you on the C3 discord server?

rudy-in commented:
bro went on holiday again.....

tareifz commented on vendor/README.md:
```diff
@@ -80,6 +80,7 @@ This is a list of grammars that Linguist selects to provide syntax highlighting
 - **C++:** [mikomikotaishi/c.tmbundle](https://github.com/mikomikotaishi/c.tmbundle)
 - **C-ObjDump:** [nanoant/assembly.tmbundle](https://github.com/nanoant/assembly.tmbundle)
 - **C2hs Haskell:** [atom-haskell/language-haskell](https://github.com/atom-haskell/language-haskell)
+- **C3:** [mikomikotaishi/c.tmbundle](https://github.com/mikomikotaishi/c.tmbundle)
```
aren't you suppose to add C3 grammar from VS Code here?

https://github.com/c3lang/vscode-c3

rudy-in commented:
morning , should i close my PR?

michabay05 commented:
Closing this PR and moving to #7516.
